src/components/ActivitiesListFilters.js

```diff
@@ -1,16 +1,27 @@
 import React from 'react';
-import {connect} from 'react-redux';
-import {setTextFilter} from '../actions/filters';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { setTextFilter } from '../actions/filters';
 
-const ActivitiesListFilters = ({ filters, dispatch }) => (
+const ActivitiesListFilters = ({ filters: { text }, dispatch }) => (
   <div>
-    <input type="text" value={filters.text} onChange={(e) => {
-      dispatch(setTextFilter(e.target.value))
-    }}/>
+    <input
+      type="text"
+      value={text}
+      onChange={e => {
+        dispatch(setTextFilter(e.target.value));
+      }}
+    />
   </div>
 );
 
-const mapStateToProps = (state) => ({
+ActivitiesListFilters.propTypes = {
+  filters: PropTypes.object,
```

Comment:
in the same way. If props are not required, make some defaults. If are required make it so:
`PropTypes.string.isRequired`. 
adding object is, simply saying a shit. There is no info what kind of object. Check `shape` in documention of prop-types and implement it.